pkg/e2etest/e2e_test.go

```diff
@@ -261,20 +276,46 @@ func testE2E(t *testing.T, datasets ...[]byte) {
 
 	// Run GoQuery and build reference results from tracking
 	resGoQuery := new(results.Result)
-	runGoQuery(t, resGoQuery, []string{
+	queryArgs := []string{
 		"-i", strings.Join(mockIfaces.Names(), ","),
 		"-e", "json",
 		"-l", time.Now().Add(time.Hour).Format(time.ANSIC),
 		"-d", tempDir,
 		"-n", strconv.Itoa(100000),
 		"-s", "packets",
 		"sip,dip,dport,proto",
-	})
-	resReference, listReference := mockIfaces.BuildResults(t, tempDir, resGoQuery)
+	}
+	dir := ""
+	if valFilterNode != nil {
+		filter := reflect.ValueOf(valFilterNode.ValFilter).Pointer()
```

Comment:
See comment about comparing functions in pkg/goDB/conditions/node/filter_direction_test.go below...

Reply:
Again, sorry for being a PITA, but I still don't think we should use reflection to compare functions (in particular if we can't guarantee correct behavior). This may work now (in this particular setup) but it's not sure it will continue to do so or even run properly e.g. on a different architecture...

Why not simply add a local iota (in this test) and a function lookup slice and then simply using the indexes / iota values to do all the comparisons (similarly for the other uses of reflection further below). Yes, you wouldn't be testing the "equality" of the _function_, but the chance of introducing / missing an error would be minimal IMHO and at the same time you would get rid of all the reflection / pointer magic? What do you think @silvanbitterli ?

Reply:
I agree with your reasoning, it's not worth introducing undefined behavior for.
I will get rid of function comparisons in the E2E tests and the direction filter unit tests.